src/controllers/petsController.js

```diff
@@ -145,6 +149,115 @@ const patchPet = async (req, res, next) => {
     });
 };
 
+const getNews = async (req, res, next) => {
+  log.debug("Calling getNews...");
+  await getAllNews(req.app.get('db'))
+    .then((dbResponse) => {
+      res.send(dbResponse);
+    })
+    .catch((e) => {
+      log.error(e);
+      res.status(500).json({ error: e.message });
+      next(e);
+    });
+};
+
+const postPetNews = async (req, res, next) => {
+  log.debug("Calling postPetNews...Verifying user inputs...");
+
+  try {
+    // check content type
+    if(!inputValidation.checkContentType(req, res)) throw new ContentTypeError();
+    // check all attrs are provided (input validation)
+    let errList = inputValidation.getMissingAttrs(res, req.body, ['newsItem'])
+    if (errList.length != 0) throw new PropRequiredError(errList);
+    // check null values
+    errList = inputValidation.getNullorEmpty(res, req.body);
+    if (errList.length != 0) throw new PropNullorEmptyError(errList);
+  } catch (err) {
+    console.log(err);
+    return res.status(err.statusCode).json({ error: err.message });
+  }
+
+  // verify that petID exists first
+  const { petID } = req.params;
+  await getPetById(req.app.get('db'), petID)
+  .then((dbResponse) => {
+    if (dbResponse.length == 0) {
+      return res.status(404).json({ error: "Pet not found." });
+    }
+    createPetNews(req.app.get('db'), petID, req.body.newsItem)
+      .then((dbResponse) => {
+        res.status(201).send(dbResponse);
+      })
+      .catch((e) => {
+        log.error(e);
+        res.status(500).json({ error: e.message });
+        next(e);
+      });
+  })
+  .catch((e) => {
+    log.error(e);
+    res.status(500).json({ error: e.message });
+    next(e);
+  });
+};
+
+const getPetNews = async (req, res, next) => {
+  log.debug("Calling getPetNews...");
+  // verify that petID exists first
```

Comment:
Is it necessary to verify petID first? If there is not petID, wouldn't `gePetNewsByPetId` just return nothing?

Reply:
Hmm this is also more like a personal preference, I prefer to return a 404 error if the given petID doesn't exist instead of returning ok with an empty list. I think it's more true to the real status and I did the same for all pets endpoints with a petID. This is unlikely to happen in the frontend though so I could remove it. What do you think?

Reply:
Yeah your reasoning makes sense, and like you said it's personal preference. It's not a super expensive query so I'm cool with leaving it in